confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSourceUtil.java

```diff
@@ -253,38 +277,37 @@ public JavaRDD<StructuredRecord> call(JavaRDD<ConsumerRecord<Object, Object>> in
     }
 
     @Override
-    public JavaRDD<StructuredRecord> call(JavaRDD<ConsumerRecord<byte[], byte[]>> input, Time batchTime) {
-      Function<ConsumerRecord<byte[], byte[]>, StructuredRecord> recordFunction = conf.getFormat() == null ?
-        new BytesFunction(batchTime.milliseconds(), conf, outputSchema) :
-        new FormatFunction(batchTime.milliseconds(), conf, outputSchema);
-      return input.map(recordFunction);
+    public JavaRDD<StructuredRecord> call(JavaRDD<ConsumerRecord<Object, Object>> input, Time batchTime) {
```

Comment:
Any reason why the type is changed from <byte[],byte[]> to <Object, Object> ? Same for the BaseFunction.

Reply:
In the `AvroFunction` static class, the message is cast to `GenericRecord` while adding the message. The compiler gives an error if the message is of the type byte array.

`GenericRecord genericRecord = (GenericRecord) message;`

Reply:
There is no direct dependence on `AvroFunction` but `RecordTranform` is using the same function super class that `AvroRecordTranform` uses.

`RecordTransform` uses either of two functions:

- `BytesFunction extends BinaryBaseFunction extends BaseFunction implements Function2`
- `FormatFunction extends BinaryBaseFunction extends BaseFunction implements Function2`

While `AvroRecordTransform` uses 

- `AvroFunction extends BaseFunction implements Function2`

And we need to define the message key and value type in the return type for this method: 
 ```
static Function2<ConsumerRecord<Object, Object>, Time, StructuredRecord>
  getRecordTransformFunction(ConfluentStreamingSourceConfig conf, Schema outputSchema) 
```
If we change the type for the other function to `<byte[], byte[]>`, then we would need to change the return type for this method accordingly as well but then it won't work for AvroFunction.